### DISCUSSION TITLE
is there way to get Endpoint result with some additional details and status code

### DISCUSSION THREAD

**Discussion by anandsubbu007** [serverpod/serverpod#184] [Q&A]
in endpoints is there a way to get endpoint result with some additional info like statuscode (200,500,401...)?
and with some other json details?

**Comment by vlidholt** [comment]
Currently, this is handled internally by the `Client` class, which will throw an exception that includes the status code if you get anything other than 20X.

Can you describe the use case for what you want to achieve?

**Reply by anandsubbu007** [reply]
if I want to give some additional details and custom message like. when I fetch for list of table row with limit. I want to let the client know that n no. of items is available for this filter. (pagination - to get total available count). 

may need to send a custom message

**Reply by anandsubbu007** [reply]
> Currently, this is handled internally by the `Client` class, which will throw an exception that includes the status code if you get anything other than 20X.
> 
> Can you describe the use case for what you want to achieve?

May i know how to get statuscode from client?

**Reply by anandsubbu007** [reply]
25a6e69661ca5bf82b2870d295da46f07b3beb9d
implemented in this PR

https://github.com/serverpod/serverpod/pull/188#issuecomment-1133666825
